### ISSUE TITLE
`pad(mode='reflect')` beyond `input.size`

### ISSUE THREAD

**Issue by OverLordGoldDragon** [pytorch/pytorch#61149]
A needed feature; I've [implemented it](https://github.com/lostanlen/kymatio/blob/fix-672/kymatio/scattering1d/backend/agnostic_backend.py#L63) for reference, but it's Python and slow.

Current limitation:

```python
torch.nn.functional.pad(torch.randn(1, 1, 128), pad=[0, 256], mode='reflect')
```

**Comment by heitorschueroff** [comment]
@OverLordGoldDragon Thank you for filing this issue. Could you please elaborate more on what feature exactly are you requesting? Thank you!

**Comment by OverLordGoldDragon** [comment]
@heitorschueroff To pad by more than input size; code I provided errors:

```python
*** RuntimeError: Argument #4: Padding size should be less than the corresponding input dimension, 
but got: padding (0, 256) at dimension 2 of input [1, 1, 128]
```

The reference I provided implements `reflect` for any pad length, and I'd port it to PyTorch but I'm guessing that'll require coding in C, which is beyond me.

**Comment by heitorschueroff** [comment]
@jbschlosser Could you review this request please?

**Comment by mruberry** [comment]
fyi @kurtamohler, who's working on torch.pad, which extends torch.nn.functional.pad's functionality.

**Comment by kurtamohler** [comment]
> fyi @kurtamohler, who's working on torch.pad, which extends torch.nn.functional.pad's functionality.

Right, we plan to support reflection padding of any size in `torch.pad`
